asdf/asdf.py

```diff
@@ -206,7 +206,7 @@ def version(self, value):
         self._user_extensions = self._process_user_extensions(self._user_extensions)
         self._plugin_extensions = self._process_plugin_extensions()
         self._extension_manager = None
-        self._extension_list = None
+        self.__extension_list = None
```

Comment:
Don't use `__` prefixes like this, this causes names to get mangled, see https://docs.python.org/3/tutorial/classes.html#private-variables, which can potentially break things for users.

You should pick a different name such as `_extension_list_` or `_private_extension_list` instead.

Reply:
Changed to `_extension_list_`.